Classes/Mapper/Event.php

```diff
@@ -174,15 +221,43 @@ public function findForAutomaticStatusChange(): Collection
      */
     public function findForRegistrationDigestEmail(): Collection
     {
-        $whereClause = 'registrations <> 0' .
-            ' AND object_type <> ' . \Tx_Seminars_Model_Event::TYPE_TOPIC .
+        $whereClause = 'registrations <> ? AND object_type <> ? AND hidden = ? AND deleted = ? ' .
             ' AND EXISTS (' .
             'SELECT * FROM tx_seminars_attendances ' .
-            'WHERE tx_seminars_attendances.deleted = 0 ' .
+            'WHERE tx_seminars_attendances.deleted = ? ' .
             ' AND tx_seminars_attendances.seminar = tx_seminars_seminars.uid' .
             ' AND tx_seminars_attendances.crdate > tx_seminars_seminars.date_of_last_registration_digest' .
             ')';
 
-        return $this->findByWhereClause($whereClause, 'begin_date ASC');
+        $sql = 'SELECT * FROM ' . $this->getTableName() . ' WHERE ' . $whereClause . ' ORDER BY begin_date ASC';
+
+        /** @var Connection $connection */
+        $connection = $this->getConnectionForTable($this->getTableName());
+        $statement = $connection->prepare($sql);
+        $statement->bindValue(1, 0, \PDO::PARAM_INT);
+        $statement->bindValue(2, \Tx_Seminars_Model_Event::TYPE_TOPIC, \PDO::PARAM_INT);
+        $statement->bindValue(3, 0, \PDO::PARAM_INT);
+        $statement->bindValue(4, 0, \PDO::PARAM_INT);
+        $statement->bindValue(5, 0, \PDO::PARAM_INT);
```

Comment:
Instead of the numbered parameters (which are prone to breakage are very hard to read), can we have named parameters instead?

Reply:
I tried first but got an SQL-error which looked like that the named parameter was not substituted. I will give it another shot and debug whats going on there.

Reply:
It seems TYPO3 doing some extra magic here with the `createNamedParameters()` function on the QueryBuilder. Doctrine DBAL, which I use here directly does not support named parameter or better to say, mysqli does not support named parameters. Please see https://github.com/doctrine/dbal/issues/2156.

We have three options here. 

1. Leave as is
2. Just use plain SQL-Query without `?` 
3. Refactor the query so that it can be build with the QueryBuilder.

Reply:
Then I opt for option 2 here. (We're hand-assembling most of the query anyway.)